prototype/src/utils/responseUtils.ts

```diff
@@ -0,0 +1,23 @@
+export function parseEntityResponse(data: any): any[] | null {
+    if (data === null || data === undefined) {
+        return null;
+    }
+
+    // Handle { body: [] } or { body: [...] } shape
+    if (data && typeof data === 'object' && 'body' in data && Array.isArray(data.body)) {
+        if (data.body.length === 0) {
+            return null;
+        }
+        return data.body;
+    }
+
+    // Handle standard array response
+    if (Array.isArray(data)) {
+        if (data.length === 0) {
+            return null;
+        }
+        return data;
+    }
+
+    return data;
```

Comment:
```suggestion
    return null;
```

In function definition it says the function returns only ` any[ ] | null ` which means it returns only an array or null. but at the end the function it returns data. so data can be any typeof (boolean, object, int etc). 

Since 'any' disable type checking in typescript its better to make the function using generics <T>, so the type can be defined separately when use the function in multiple places.